tests/cxx/using_overload.cc

```diff
@@ -0,0 +1,30 @@
+//===--- using_overload.cc - test input file for iwyu ---------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
```

Comment:
Please add comments in tests what exactly the test is supposed to catch. For example,

> Tests that if we encounter using overloaded function, iwyu recommends to keep only files with actually used function signatures.

And maybe add a test case where you use both `swap(vector, vector)` and `swap(list, list)`.